internal/pipeline/pipeline_function.go

```diff
@@ -21,70 +21,17 @@ import (
 	"io/ioutil"
 	"os"
 	"path"
-	"regexp"
-	"strings"
 
 	"github.com/GoogleContainerTools/kpt/internal/pipeline/runtime"
+	"github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
 	"sigs.k8s.io/kustomize/kyaml/kio"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-// Function defines an item in the pipeline function list
-type Function struct {
-	// `Image` is the path of the function container image
-	// Image name can be a "built-in" function: kpt can be configured to use a image
-	// registry host-path that will be used to resolve the full image path in case
-	// the image path is missing (Defaults to gcr.io/kpt-functions-trusted).
-	// For example, the following resolves to gcr.io/kpt-functions-trusted/patch-strategic-merge.
-	//		image: patch-strategic-merge
-	Image string `yaml:"image,omitempty"`
-
-	// `Config` specifies an inline k8s resource used as the function config.
-	// Config, ConfigPath, and ConfigMap fields are mutually exclusive.
-	Config yaml.Node `yaml:"config,omitempty"`
-
-	// `ConfigPath` specifies a relative path to a file in the current directory
-	// containing a K8S resource used as the function config. This resource is
-	// excluded when resolving 'sources', and as a result cannot be operated on
-	// by the pipeline.
-	ConfigPath string `yaml:"configPath,omitempty"`
-
-	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
-	ConfigMap map[string]string `yaml:"configMap,omitempty"`
-}
-
-// Validate will validate all fields in function.
-func (f *Function) Validate() error {
-	err := validateFunctionName(f.Image)
-	if err != nil {
-		return fmt.Errorf("'image' is invalid: %w", err)
-	}
-
-	var configFields []string
-	if f.ConfigPath != "" {
-		if err := validatePath(f.ConfigPath); err != nil {
-			return fmt.Errorf("'configPath' %q is invalid: %w", f.ConfigPath, err)
-		}
-		configFields = append(configFields, "configPath")
-	}
-	if len(f.ConfigMap) != 0 {
-		configFields = append(configFields, "configMap")
-	}
-	if !isNodeZero(&f.Config) {
-		configFields = append(configFields, "config")
-	}
-	if len(configFields) > 1 {
-		return fmt.Errorf("following fields are mutually exclusive: 'config', 'configMap', 'configPath'. Got %q",
-			strings.Join(configFields, ", "))
-	}
-
-	return nil
-}
-
-// runner returns a fnRunner from the image and configs of
+// newFnRunner returns a fnRunner from the image and configs of
 // this function.
-func (f *Function) runner() (kio.Filter, error) {
-	config, err := f.config()
+func newFnRunner(f *v1alpha2.Function, pkgPath string) (kio.Filter, error) {
```

Comment:
Do we need to do a validation for the config name for example?

Reply:
Do you mean validating is the config file name a valid file name?